usaspending_api/references/management/commands/load_psc.py

```diff
@@ -1,81 +1,176 @@
+                usecols=[
+                    "PSC CODE",
+                    "START DATE",
+                    "END DATE",
+                    "PRODUCT AND SERVICE CODE NAME",
+                    "PRODUCT AND SERVICE CODE FULL NAME (DESCRIPTION)",
+                    "PRODUCT AND SERVICE CODE INCLUDES",
+                    "PRODUCT AND SERVICE CODE EXCLUDES",
+                    "PRODUCT AND SERVICE CODE NOTES",
+                ],
+                parse_dates=["START DATE", "END DATE"],
+                dtype={"PSC CODE": str},
+            )
+            # Rename columns to match PSC model fields
+            .rename(
+                columns={
+                    "PSC CODE": "code",
+                    "START DATE": "start_date",
+                    "END DATE": "end_date",
+                    "PRODUCT AND SERVICE CODE FULL NAME (DESCRIPTION)": "full_name",
+                    "PRODUCT AND SERVICE CODE NAME": "description",
+                    "PRODUCT AND SERVICE CODE INCLUDES": "includes",
+                    "PRODUCT AND SERVICE CODE EXCLUDES": "excludes",
+                    "PRODUCT AND SERVICE CODE NOTES": "notes",
+                }
+            )
+            .loc[
+                lambda df: (
+                    # Remove duplicate codes keeping most recent start_date
+                    ~df.sort_values(by=["code", "start_date"]).duplicated(subset=["code"], keep="last")
+                    # Remove rows with missing descriptions
+                    & df["description"].notnull()
+                )
+            ]
+            # Convert start date and end date to dates
+            .assign(start_date=lambda df: df["start_date"].dt.date, end_date=lambda df: df["end_date"].dt.date)
+            # Add length column
+            .assign(length=lambda df: df["code"].astype(str).str.len())
+            # Replace missing values with None
+            .astype(object)
+            .where(lambda df: df.notna(), None)
+            .to_dict(orient="records")
+        )
+        with transaction.atomic():
+            old = {psc.code: psc for psc in PSC.objects.all()}
+            PSC.objects.all().delete()
+            created = PSC.objects.bulk_create(PSC(**new_psc) for new_psc in new_pscs)
+            new = {psc.code: psc for psc in created}
+            result = compare_codes(old, new)
         if update:
             update_lengths()
             logger.log(20, "Updated PSC codes.")
-    except IOError:
-        logger.error("Could not open file {}".format(fullpath))
-
-
-def check_start_end_dates(psc, start_date, end_date):
-    if psc.start_date:
-        if start_date:
-            if psc.start_date < start_date.date():
-                psc.start_date = start_date
-    else:
-        psc.start_date = start_date
-    if psc.end_date:
-        if end_date:
-            if psc.end_date < end_date.date():
-                psc.end_date = end_date
-
-    else:
-        psc.end_date = end_date
+    except IOError as e:
```

Comment:
Since the result is the same regardless of the error is there any benefit to having the different exceptions called out?

Reply:
Since we're not doing anything to handle the error and if we want it to raise the error, should we just remove the try except entirely and let it fail and raise the error if there is a problem?

Reply:
I don't see why not. Not sure why this would have silenced the error before.